src/dune_rules/ml_sources.ml

```diff
@@ -294,14 +310,46 @@ let libs_and_exes dune_file ~dir ~scope ~lookup_vlib ~modules =
       | _ -> Memo.return List.Skip)
   >>| List.filter_partition_map ~f:Fun.id
 
+let melange_emits dune_file ~dir ~scope ~modules =
+  (* TODO: this function could be performed in parallel as part of libs_and_exes or some variation *)
+  Memo.parallel_map dune_file.stanzas ~f:(fun stanza ->
+      match (stanza : Stanza.t) with
+      | Melange_emit mel ->
+        let modules =
+          let modules =
+            Modules_field_evaluator.eval_no_buildable ~modules
+              ~stanza_loc:mel.loc ~modules_field:mel.entries
+              ~modules_without_implementation:Ordered_set_lang.standard
+              ~kind:Modules_field_evaluator.Exe_or_normal_lib
+              ~private_modules:Ordered_set_lang.standard
+          in
+          let project = Scope.project scope in
+          (* TODO: melange should check some other config flag, not exe one *)
+          if Dune_project.wrapped_executables project then
```

Comment:
Even better would be if melange didn't have this configuration option at all and stuck to wrapping. The option exists in dune only because of backwards compatibility.

Reply:
Removed in https://github.com/ocaml/dune/pull/6349/commits/b93c5aaae85e4c542e8e4a7df61bff853f6c8d81.